File: 1.two-sum.java
/*
 * @lc app=leetcode id=1 lang=java
 *
 * [1] Two Sum
 */
class Solution {
    public int[] twoSum(int[] nums, int target) {
        if (nums == null) return new int[] {};
        Map<Integer, Integer> idxMap = new HashMap<>();
        for (int i = 0; i < nums.length; ++i) {
            if (idxMap.containsKey(nums[i]))
                return new int[] {idxMap.get(nums[i]), i};
            else
                idxMap.put(target-nums[i], i);
        }
        return new int[] {};
    }
}

File: go/1.two-sum.go
/*
 * @lc app=leetcode id=1 lang=golang
 *
 * [1] Two Sum
 */

// @lc code=start
func twoSum(nums []int, target int) []int {
	loc := make(map[int]int)
	for i, v := range nums {
		if j, b := loc[v]; b {
			return []int{j, i}
		} else {
			loc[target-v] = i
		}
	}
	return nil
}

// @lc code=end

